### ISSUE TITLE
Add Ordering Tests For PackedVector

### ISSUE THREAD

**Issue by tomspilman** [mono/MonoGame#4685]
Our tests for packed vector types only test against a zero and one types... it does not include tests with different values for each component.  This means the tests do not correctly detect errors in the packing order.

We should add a test for packing order to each of the PackedVectorTests types in:

https://github.com/mono/MonoGame/blob/develop/Test/Framework/PackedVectorTest.cs

Use the test in the `Rgba64` test as the example as to how to form the new tests:

https://github.com/mono/MonoGame/blob/develop/Test/Framework/PackedVectorTest.cs#L85

While we are at this the coverage of the PackedVectorTest tests is pretty poor.  We should expand these to include testing:

   - `IPackedVector.ToVector4()` across all types.
   - Equality methods across all types.
   - `HalfSingle`
   - `HalfVector2`
   - `HalfVector4`

**Comment by flibitijibibo** [comment]
These aren't tests, but they may be helpful to pass them:

After seeing #4680 and doing some comparisons to XNA, it turns out basically all of the PackedVectors were wrong! On top of ordering issues, it turns out rounding issues are a problem as well. I've fixed them in FNA (the changes should work just fine in MG):

https://github.com/FNA-XNA/FNA/commit/fdc2171f507ea81f6d721da7ee4e7218c12b145d
https://github.com/FNA-XNA/FNA/commit/3d3582afb98e3be94d7a704bda60c95e2b8b2b32

I referred to how MS documented the process for their packed vectors in DXMath:

https://msdn.microsoft.com/en-us/library/windows/desktop/ee415635%28v=vs.85%29.aspx

Clamp, round, bitshift. The exception is Normalized vectors which apparently operate on [-1, 1] ranges unlike the DXMath variants.
